lib/parse-mvn.js

```diff
@@ -77,31 +77,42 @@ function assemblePackage(source, projectDeps, parent, withDev) {
 
 function createPackage(pkgStr, parent) {
   var range = getConstraint(pkgStr);
+
   if (range) {
     pkgStr = pkgStr.substring(0, pkgStr.indexOf(' '));
   }
+
   var parts = pkgStr.split(':');
+  var version = parts[3];
+  var groupId = parts[0];
+  var artifactId = parts[1];
   var result = {
-    groupId: parts[0],
-    artifactId: parts[1],
+    groupId: groupId,
+    artifactId: artifactId,
     packaging: parts[2],
-    version: parts[3],
+    version: version,
     name: parts[0] + ':' + parts[1],
     dependencies: {},
   };
-  var selfPkg = parts[0] + ':' + parts[1] + '@' + parts[3];
+  var selfPkg = groupId + ':' + artifactId + '@' + version;
```

Comment:
And then move this var def to happen after the `parts.length >= 5` check, and rely on `result.groupId + ':' + result.artifactId + '@' + result.version`